generated/fig/tendermint/codec/v1/codec.proto

```diff
@@ -0,0 +1,802 @@
+message LightBlock {
+  SignedHeader signed_header = 1; // cannot be nil
+  ValidatorSet validator_set = 2; // cannot be nil
+}
+
+message LightClientAttackEvidence {
+  LightBlock         conflicting_block    = 1;
+  int64              common_height        = 2;
+  repeated Validator byzantine_validators = 3;
+  int64              total_voting_power   = 4;
+  Timestamp          timestamp            = 5 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+}
+
+message PublicKey {
+  option (gogoproto.compare) = true;
+  option (gogoproto.equal)   = true;
+
+  oneof sum {
+    bytes ed25519   = 1;
+    bytes secp256k1 = 2;
+  }
+}
+
+message PartSetHeader {
+  uint32 total = 1;
+  bytes  hash  = 2; // hash 256 32 bytes
+}
+
+message ResponseBeginBlock {
+  repeated Event events = 1 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
+}
+
+message ResponseEndBlock {
+  repeated ValidatorUpdate validator_updates       = 1;
+  ConsensusParams          consensus_param_updates = 2;
+  repeated Event           events                  = 3;
+}
+
+message ResponseDeliverTx {
+  uint32         code       = 1;
+  bytes          data       = 2;
+  string         log        = 3;
+  string         info       = 4;
+  int64          gas_wanted = 5;
+  int64          gas_used   = 6;
+  repeated Event events     = 7 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
+  string         codespace  = 8;
+}
+
+message SignedHeader {
+  Header header = 1;
+  Commit commit = 2;
+}
+
+message Timestamp {
+  int64 seconds = 1;
+  int32 nanos   = 2;
+}
+
+message TxResult {
+  uint64            height = 1;
+  uint32            index  = 2;
+  Tx                tx     = 3;
+  ResponseDeliverTx result = 4;
+  bytes             hash   = 5;
+}
+
+message Validator {
+  bytes     address           = 1;
+  PublicKey pub_key           = 2 [(gogoproto.nullable) = false]; // must be > 0
+  int64     voting_power      = 3;
+  int64     proposer_priority = 4;
+}
+
+message ValidatorParams {
+  option (gogoproto.populate) = true;
+  option (gogoproto.equal)    = true;
+
+  repeated string pub_key_types = 1;
+}
+
+message ValidatorSet {
+  repeated Validator validators         = 1; // cannot be empty or nil
+  Validator          proposer           = 2; // cannot be nil
+  int64              total_voting_power = 3;
+}
+
+message ValidatorUpdate {
+  bytes     address = 1;
+  PublicKey pub_key = 2 [(gogoproto.nullable) = false];
+  int64     power   = 3;
+}
+
+message VersionParams {
+  option (gogoproto.populate) = true;
+  option (gogoproto.equal)    = true;
+
+  uint64 app_version = 1;
+}
+
+//------------------------------------------------------------------------
+// imported from fig/tendermint/transform/v1/transform.proto
+//
+
+message EventFilter {
+  repeated string event_types = 1;
+}
+
+//------------------------------------------------------------------------
+// imported from protoc-gen-openapiv2/options/openapiv2.proto
+//
```

Comment:
Are the types below used in `cosmos.Tx` (but not in messages)? If not, we don't need them in `codec.proto`.